src/libraries/Microsoft.PowerFx.Interpreter/Marshal/ObjectMarshallerProvider.cs

```diff
@@ -50,6 +60,30 @@ public bool TryGetMarshaller(Type type, TypeMarshallerCache cache, int maxDepth,
 
             var fxType = new RecordType();
 
+            fxType = GetProperties(type, cache, maxDepth, mapping, fxType);
```

Comment:
>mapping [](http://example.com/codeflow?start=58&length=7)

so this populates the mapping (as a side-effect) and then returns the fxType? effectively 2 out params.

Reply:
Yeah I just extracted the existing logic out of the TryGetMarshaller into its own (virtual) method so it can be extended.